### PR TITLE
fix(v2): add base url to opensearch.xml

### LINKED ISSUE #4151: opensearch.xml template does not respect `baseUrl` option
On sites which are built using [the `baseUrl` option](https://v2.docusaurus.io/docs/docusaurus.config.js/#baseurl), the `<link rel="search">` tag in the `head` is generated correctly, and the `opensearch.xml` file is generated in _the right location_ wrt the supplied `baseUrl`, but __the contents of `opensearch.xml` file are incorrect__.

Consider [Facebook Metro](https://facebook.github.io/metro/), which is cited in the docs as an example of a `baseUrl` deployment. The `link` tag on those pages is correct...

```html
<link rel="search" type="application/opensearchdescription+xml" title="Metro" href="/metro/opensearch.xml">
```

...but the `opensearch.xml` contents are not. The `Image`, `Url` and `moz:SearchForm` tags all fail to include the `baseUrl` as part of their urls, which breaks the function, and/or causes browsers to register an incorrect search URL for the entire domain `facebook.github.io`.

```xml
<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/" xmlns:moz="http://www.mozilla.org/2006/browser/search/">
<ShortName>Metro</ShortName>
<Description>Search Metro</Description>
<InputEncoding>UTF-8</InputEncoding>
<Image width="16" height="16" type="image/x-icon">https://facebook.github.io/img/favicon.png</Image>
<Url type="text/html" method="get" template="https://facebook.github.io/search?q={searchTerms}"/>
<Url type="application/opensearchdescription+xml" rel="self" template="https://facebook.github.io/opensearch.xml"/>
<moz:SearchForm>https://facebook.github.io</moz:SearchForm>
</OpenSearchDescription>
```

In the case of [Facebook Metro](https://facebook.github.io/metro/), the `opensearch.xml` file should look like this:

```xml
<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/" xmlns:moz="http://www.mozilla.org/2006/browser/search/">
<ShortName>Metro</ShortName>
<Description>Search Metro</Description>
<InputEncoding>UTF-8</InputEncoding>
<Image width="16" height="16" type="image/x-icon">https://facebook.github.io/metro/img/favicon.png</Image>
<Url type="text/html" method="get" template="https://facebook.github.io/metro/search?q={searchTerms}"/>
<Url type="application/opensearchdescription+xml" rel="self" template="https://facebook.github.io/metro/opensearch.xml"/>
<moz:SearchForm>https://facebook.github.io</moz:SearchForm>
</OpenSearchDescription>
```

### PR DESCRIPTION
**Pull request by lex111**
## Motivation

Fixes #4151.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

Set `baseUrl` and check out opensearch.xml file.

Preview - https://deploy-preview-4154--docusaurus-2.netlify.app/classic/opensearch.xml

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: packages/docusaurus-theme-search-algolia/src/index.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ function theme(context) {
           path.join(outDir, OPEN_SEARCH_FILENAME),
           renderOpenSearchTemplate({
             title,
-            url,
+            url: url + baseUrl,
             favicon: normalizeUrl([url, favicon]),
```
Comment (slorber) [inline packages/docusaurus-theme-search-algolia/src/index.js:63]:
```suggestion
            favicon: normalizeUrl([url, baseUrl, favicon]),
```

File: packages/docusaurus-theme-search-algolia/src/templates/opensearch.js [hunk 2]
```diff
@@ -14,7 +14,7 @@ module.exports = `
   <InputEncoding>UTF-8</InputEncoding>
   <Image width="16" height="16" type="image/x-icon"><%= it.favicon %></Image>
   <Url type="text/html" method="get" template="<%= it.url %>/search?q={searchTerms}"/>
```
Comment (slorber) [inline packages/docusaurus-theme-search-algolia/src/templates/opensearch.js:16]:
```suggestion
  <Url type="text/html" method="get" template="<%= it.url %>search?q={searchTerms}"/>
```

**Review by slorber — CHANGES_REQUESTED** [review]
Thanks

The preview using baseUrl show some missing baseurl for favicon + a double slash:
https://deploy-preview-4154--docusaurus-2.netlify.app/classic/opensearch.xml

**Comment by lex111** [comment]
@slorber missed it, thanks :heavy_check_mark:

**Pull request merged**